src/kudu/consensus/log_cache.cc

```diff
@@ -369,50 +301,21 @@ Status LogCache::AppendOperations(const vector<ReplicateRefPtr>& msgs,
   int64_t compressed_size = 0;
   int64_t uncompressed_size = 0;
   vector<CacheEntry> entries_to_insert;
-  entries_to_insert.reserve(msgs.size());
+  entries_to_insert.reserve(msg_wrappers.size());
+
+  for (const auto& msg_wrapper : msg_wrappers) {
+    auto msg = msg_wrapper.Msg();
+    auto compressed_msg = msg_wrapper.CompressedMsg();
 
-  bool found_compressed_msgs = false;
-  for (const auto& msg : msgs) {
     CacheEntry e;
-    e.mem_usage = 0;
     e.msg_size = msg->get()->SpaceUsedLong();
 
-    const auto op_type = msg->get()->op_type();
-    bool is_compressed =
-      (msg->get()->write_payload().compression_codec() != NO_COMPRESSION);
     uncompressed_size += msg->get()->write_payload().payload().size();
```

Comment:
This part was weird in the orig code, we were incrementing uncompressed_size metric even when the msg is compressed. This was even expected by some unit tests on the plugin side.

Reply:
looks wrong to me as well.